ocaml/libs/stunnel/stunnel.ml

```diff
@@ -35,27 +35,32 @@ let stunnel_logger = ref ignore
 let timeoutidle = ref None
 
 let init_stunnel_path () =
-  try cached_stunnel_path := Some (Unix.getenv "XE_STUNNEL")
-  with Not_found ->
-    let choices =
-      [
-        "/opt/xensource/libexec/stunnel/stunnel"
-      ; "/usr/sbin/stunnel4"
-      ; "/usr/sbin/stunnel"
-      ; "/usr/bin/stunnel4"
-      ; "/usr/bin/stunnel"
-      ]
-    in
-    let rec choose l =
-      match l with
-      | [] ->
-          raise Stunnel_binary_missing
-      | p :: ps -> (
-        try Unix.access p [Unix.X_OK] ; p with _ -> choose ps
+  cached_stunnel_path :=
+    Some
+      ( match Sys.getenv_opt "XE_STUNNEL" with
+      | Some x ->
+          x
+      | None ->
+          let choices =
+            [
+              "/opt/xensource/libexec/stunnel/stunnel"
+            ; "/usr/sbin/stunnel4"
+            ; "/usr/sbin/stunnel"
+            ; "/usr/bin/stunnel4"
+            ; "/usr/bin/stunnel"
+            ]
+          in
+          let rec choose l =
```

Comment:
This could be implemented using `List.find_opt`; the predicate would use `Unix.access` to test the element.